autoload/db/adapter/bigquery.vim

```diff
@@ -0,0 +1,81 @@
+if !executable('bq')
+  echom 'bq command is not available.'
+  let g:autoloaded_db_bigquery = 0
+endif
+
+if exists('g:autoloaded_db_bigquery')
+  finish
+endif
+
+let g:autoloaded_db_bigquery = 1
+
+function! s:command_for_url(params, action) abort
+  let cmd = ['bq']
+  let subcmd = [a:action]
+
+  for [k, v] in items(a:params)
+    let op = ' --'.k.'='.v
+
+    if len(matchstr(k, 'legacy'))
+        " FIXME: parse subcmd options
+        call add(subcmd, op)
+    else
+        " parse global options
+        call add(cmd, op)
+    endif
+
+  endfor
+
+  return join(cmd + subcmd)
```

Comment:
You're building a string here, only to `split()` it back to an list in `#interactive()`. You should build a list to begin with.